extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js

```diff
@@ -0,0 +1,229 @@
+      this.playerReadyResolver_ = resolve;
+    });
+
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    this.applyFillContent(iframe);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+
+    this.element.appendChild(iframe);
+
+    installVideoManagerForDoc(this.element);
+    videoManagerForDoc(this.element).register(this);
+  }
+
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+
+    const mediaId = user().assert(
+      this.element.getAttribute('data-mediaid'),
+      'The data-mediaid attribute is required for <amp-nexxtv-player> %s',
+      this.element);
+
+    const client = user().assert(this.element.getAttribute('data-client'),
+      'The data-client attribute is required for <amp-nexxtv-player> %s',
+      this.element);
+
+    const start = this.element.getAttribute('data-seek-to') || 0;
+    const mode = this.element.getAttribute('data-mode') || 'static';
+    const streamtype = this.element.getAttribute('data-streamtype') || 'video';
+    const origin = this.element.getAttribute('data-origin')
+      || 'https://embed.nexx.cloud/';
+
+    let src = '';
+    src += origin;
+
+    if (streamtype !== 'video') {
+      src += `${encodeURIComponent(streamtype)}/`;
+    }
+
+    src += `${encodeURIComponent(client)}/`;
+    src += `${encodeURIComponent(mediaId)}`;
+    src += `?start=${encodeURIComponent(String(start))}`;
+    src += `&datamode=${encodeURIComponent(mode)}&amp=1`;
+
+    return this.videoIframeSrc_ = src;
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.iframe_.src = this.getVideoIframeSrc_();
+
+    this.unlistenMessage_ = listen(this.iframe_,'message', event => {
+      this.handleNexxMessages_(event);
+    });
+
+    return this.loadPromise(this.iframe_)
+      .then(() => {
+        this.element.dispatchCustomEvent(VideoEvents.LOAD);
+        this.playerReadyResolver_(this.iframe_);
+      });
+  }
+
+  pauseCallback() {
+    if (this.iframe_) {
+      this.pause();
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+
+    return true;
+  }
+
+  sendCommand_(command) {
+    this.iframe_.contentWindow./*OK*/postMessage(JSON.stringify({
+      'cmd': command,
+    }), '*');
+  };
+
+  // emitter
+  handleNexxMessages_(event) {
+    const data = isObject(event.data) ? event.data : tryParseJson(event.data);
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+
+    if (data.cmd == 'onload') {
+      this.element.dispatchCustomEvent(VideoEvents.LOAD);
+      this.playerReadyResolver_(this.iframe_);
```

Comment:
Bug: This will already be resolved by `#layoutCallback`. Are we supposed to wait until we receive the onload?